android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java

```diff
@@ -128,6 +126,7 @@ public class CustomTwilioVideoView extends View implements LifecycleEventListene
     private String cameraType = "";
     private boolean enableH264Codec = false;
     private String videoTrackName = "camera";
+    private static CustomVideoProcessor videoProcessor = new CustomVideoProcessor();
```

Comment:
Did you mean for this to be `static`? Unless there's a need for the `CustomVideoProcessor` to be a singleton, it's better to have an instance per VideoView.

Reply:
Also, you're not reassigning `videoProcessor` in this class, so you should make it `final`.

Reply:
Yeah, it needed to be static since its called within the static `registerThumbnailView` method